cgen error when returning fixed size array from unsafe function

### Describe the bug

A function call to an unsafe function returning a fixed size array produces a c compiler error, when wrapped with a unsafe block.

### Reproduction Steps

The following code will not compile.
```v
@[unsafe]
fn test() [5]int {
	return [5]int{}
}

fn main() {
	foo := unsafe { test() }

	println(foo)
}
```

### Expected Behavior

Compiles with no errors.

### Current Behavior

```
======== Output of the C Compiler (/home/lars/v/thirdparty/tcc/tcc.exe) ========
/tmp/v_1000/test.01JJ5CVRNZN5KWEVYH7S4GY15M.tmp.c:5896: error: '{' expected (got ";")
================================================================================
======== Output of the C Compiler (cc) ========
/tmp/v_1000/test.01JJ5CVRNZN5KWEVYH7S4GY15M.tmp.c: In function 'main__main':
/tmp/v_1000/test.01JJ5CVRNZN5KWEVYH7S4GY15M.tmp.c:5896:33: error: invalid initializer
 5896 |         Array_fixed_int_5 foo = main__test().ret_arr;
      |                                 ^~~~~~~~~~
===============================================
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.9 4e68a86

### Environment details (OS name and version, etc.)

|V full version      |V 0.4.9 d680c42.4e68a86
|:-------------------|:-------------------
|OS                  |linux, "Void Linux"
|Processor           |12 cpus, 64bit, little endian, AMD Ryzen 5 5600G with Radeon Graphics
|Memory              |4.83GB/15.52GB
|                    |
|V executable        |/home/lars/v/v
|V last modified time|2025-01-21 21:39:55
|                    |
|V home dir          |OK, value: /home/lars/v
|VMODULES            |OK, value: /home/lars/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |contains spaces, value: /home/lars/Robotron z1013/Pixel Graphics v3/generator
|                    |
|Git version         |git version 2.48.1
|V git status        |weekly.2024.36-1031-g4e68a860
|.git/config present |true
|                    |
|cc version          |cc (GCC) 13.2.0
|gcc version         |gcc (GCC) 13.2.0
|clang version       |clang version 17.0.6
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.39

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -468,10 +468,10 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 		unaliased_right_sym := g.table.final_sym(unwrapped_val_type)
 		is_fixed_array_var := !g.pref.translated && unaliased_right_sym.kind == .array_fixed
 			&& val !is ast.ArrayInit
-			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr, ast.InfixExpr, ast.UnsafeExpr]
+			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr, ast.InfixExpr]
 			|| (val is ast.CastExpr && val.expr !is ast.ArrayInit)
 			|| (val is ast.PrefixExpr && val.op == .arrow)
-			|| (val is ast.UnsafeExpr && val.expr in [ast.SelectorExpr, ast.Ident]))
+			|| (val is ast.UnsafeExpr && val.expr in [ast.SelectorExpr, ast.Ident, ast.CallExpr]))
 		g.is_assign_lhs = true
 		g.assign_op = node.op
 

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -468,7 +468,7 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 		unaliased_right_sym := g.table.final_sym(unwrapped_val_type)
 		is_fixed_array_var := !g.pref.translated && unaliased_right_sym.kind == .array_fixed
 			&& val !is ast.ArrayInit
-			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr, ast.InfixExpr]
+			&& (val in [ast.Ident, ast.IndexExpr, ast.CallExpr, ast.SelectorExpr, ast.DumpExpr, ast.InfixExpr, ast.UnsafeExpr]
 			|| (val is ast.CastExpr && val.expr !is ast.ArrayInit)
 			|| (val is ast.PrefixExpr && val.op == .arrow)
 			|| (val is ast.UnsafeExpr && val.expr in [ast.SelectorExpr, ast.Ident]))

--- a/vlib/v/tests/builtin_arrays/array_fixed_from_unsafe_test.v
+++ b/vlib/v/tests/builtin_arrays/array_fixed_from_unsafe_test.v
@@ -1,0 +1,10 @@
+@[unsafe]
+fn test() [5]int {
+	return [5]int{}
+}
+
+fn test_main() {
+	foo := unsafe { test() }
+	assert foo.len == 5
+	assert foo == unsafe { test() }
+}

